addons/stock/models/stock_move_line.py

```diff
@@ -24,7 +24,8 @@ class StockMoveLine(models.Model):
         help="Change to a better name", index=True)
     company_id = fields.Many2one('res.company', string='Company', readonly=True, required=True, index=True)
     product_id = fields.Many2one('product.product', 'Product', ondelete="cascade", check_company=True)
-    product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True)
+    product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True, domain="[('category_id', '=', product_uom_category_id)]")
```

Comment:
Problem might be kinda similar to previous one: if product_id is not set, this will only allow selecting product_uom_id with category False rather than all of them, don't know if that's expected.